### DISCUSSION TITLE
Autofill username with Social login (Google source)

### DISCUSSION THREAD

**Discussion by RobinFrcd** [goauthentik/authentik#4432] [Q&A] [answered]
Hi,
I've enabled the Google source enrollment, but I'd like to know if it's possible to disable the username prompt and automatically use the email address ?

I've tried to delete the `default-source-enrollment-prompt` stage, but it's clearly not the solution.

Is there any solution to do that ?

Thanks,

**Comment by lrobinot** [comment]
I have the same question, but as I know the user names and emails, I provisionned, throuh Api, the users.

**Comment by MaximeWewer** [comment, marked as answer]
I think this answser can help you #3709

**Reply by BeryJu** [reply]
This has also now been added to the docs https://goauthentik.io/integrations/sources/google/#username-mapping

**Reply by RobinFrcd** [reply]
I think the part about `Edit Prompts` is missing in the docs. Other than that, it works great ! Thanks !

**Reply by BeryJu** [reply]
The prompt editing part is not required

**Reply by RobinFrcd** [reply]
If I don't do this step, username is still prompted and only the place holder is filled.
![image](https://user-images.githubusercontent.com/29704178/215544922-31a19d34-6846-477f-ba2e-5e3c8dbaf78f.png)

**Reply by BeryJu** [reply]
Oh, the can you try with the policy from the docs returning `False` instead of `True`?

**Reply by RobinFrcd** [reply]
Still the same behavior
![image](https://user-images.githubusercontent.com/29704178/215546454-ac77966b-99dd-4188-a13a-e0f95a2de950.png)

**Reply by BeryJu** [reply]
can you try returning `False` and in addition also swapping the order of the policies?

**Reply by RobinFrcd** [reply]
~~I'm getting `Request has been denied.` `Unknown error`~~

My bad, I waited a bit and now it works ! Seems like a worker's job cleaned up something ! Thanks !

**Reply by BeryJu** [reply]
Updated the website to reflect this order requirement https://goauthentik.io/integrations/sources/google/#username-mapping
